colossalai/inference/config.py

```diff
@@ -176,6 +178,12 @@ def _verify_config(self) -> None:
             self.dtype in _ALLOWED_DTYPES
         ), f"Expected dtype to be in {_ALLOWED_DTYPES} but found an unknown dtype: {self.dtype}"
 
+        if self.kv_cache_dtype:
+            assert (
+                self.use_cuda_kernel and self.kv_cache_dtype == "fp8"
+            ), f"FP8 kv_cache is only supported with use_cuda_kernel open now"
+            self.kv_cache_dtype = torch.uint8
```

Comment:
Consider the condition of a user inputting something like `kv_cache_dtype="fp16"` or `kv_cache_dtype="fpp16-type"`, there won't be any warning or error message popping out. In common styling, `kv_cache_dtype` can be assigned and verified by:
1. provide a default value as the same as `dtype` 
2. identify invalid input args (warning and fall back to default value)
3. additional assertion
4. assignment

Reply:
I think may it will be wrong? 
![image](https://github.com/hpcaitech/ColossalAI/assets/28504079/7b162475-5d98-40a2-8627-f8c5a2268390)